components/eam/src/physics/cam/physpkg.F90

```diff
@@ -1340,6 +1343,24 @@ subroutine phys_run2(phys_state, ztodt, phys_tend, pbuf2d,  cam_out, &
     call gmean_mass ('after tphysac FV:WET)', phys_state)
 #endif
 
+    !
+    ! Check for carbon conservation
+    !
+    do c = begchunk, endchunk
+       call get_total_carbon(phys_state(c), 'wet')
+    end do
+    call print_global_carbon_diags(phys_state, ztodt, nstep)
+    do c = begchunk, endchunk
```

Comment:
does this block need to be wrapped in if (co2_transport) like other bocks using new C diagnostics variables?

Reply:
That's a good idea.  While the calls to `get_total_carbon` and `print_global_carbon_diags` will return immediately if co2_transport is false, we would still be setting `tc_prev`, `tc_init`, `tc_mnst` needlessly.  I will add in the `co2_transport` if block.